keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -501,15 +501,47 @@ kpxc.prepareCredentials = async function() {
         return;
     }
 
-    if (kpxc.settings.autoFillSingleEntry && kpxc.credentials.length === 1) {
+    if (kpxc.credentials.length === 1) {
+      if (kpxc.settings.autoFillSingleEntry) {
         kpxcFill.fillFromAutofill();
         return;
+      }
+
+      // Check for site preference overrides
+      const sitePreference = kpxc.retrieveSitePreference();
+      if (sitePreference !== null && sitePreference !== undefined) {
+        if (sitePreference.autoFillCredentials === true) {
+          kpxcFill.fillFromAutofill();
+          return;
+        }
+      }
     }
 
     kpxc.initLoginPopup();
     kpxc.initAutocomplete();
 };
 
+kpxc.retrieveSitePreference = function() {
```

Comment:
Instead of these checks you could just update values to temporary variables under `kpxc.` in here: https://github.com/keepassxreboot/keepassxc-browser/blob/develop/keepassxc-browser/content/keepassxc-browser.js#L771